doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/examples/ml-example.md

```diff
@@ -1,10 +1,253 @@
+```shell
+# Starting from the parent directory of cloned Ray master,
+pushd ray/doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/configs/
+kubectl apply -f xgboost-benchmark.yaml
+popd
+```
+
+A Ray head pod and 9 Ray worker pods will be created.
+
+### Deploying an autoscaling Ray cluster
+
+This option is most appropriate if you have set up an autoscaling Kubernetes
+node pool or group.
+
+We recommend taking a look at the config file applied in the following command.
+```shell
+# Starting from the parent directory of cloned Ray master,
+pushd ray/doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/configs/
+kubectl apply -f xgboost-benchmark-autoscaler.yaml
+popd
+```
+
+One Ray head pod will be created. Once the workload starts, the Ray autoscaler will trigger
+creation of Ray worker pods. Kubernetes autoscaling will then create nodes to place the Ray pods.
+
+
+## Running the workload
+
+To observe the startup progress of the Ray head pod, run the following command.
+```shell
+# If you're on MacOS, first `brew install watch`.
+watch -n 1 kubectl get pod
+```
+
+Once the Ray head pod enters `Running` state, we are ready to execute the XGBoost workload.
+We will use {ref}`Ray Job Submission<jobs-overview>` to kick off the workload.
+
+### Connect to the cluster.
+
+First, we connect to the Job server. Run the following blocking command
+in a separate shell.
+```shell
+kubectl port-forward service/raycluster-xgboost-benchmark-head-svc 8265:8265
+```
+
+### Submit the workload.
+
+We'll use the {ref}`Ray Job Python SDK<ray-job-sdk>` to submit the XGBoost workload.
+
+```{literalinclude} ../doc_code/xgboost_submit.py
+:language: python
+```
+
+To submit the workload, run the above Python script.
+The script is available in the Ray repository.
+
+```shell
+# From the parent directory of cloned Ray master.
+pushd ray/doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/doc_code/
+python xgboost_submit.py
+popd
+```
+
+### Observe progress.
+
+The benchmark may take up to 30 minutes to run.
+Use the following tools to observe its progress.
+
+#### Job logs
+
+To follow the job's logs, use the command printed by the above submission script.
+```shell
+# Subsitute the Ray Job's submission id.
+ray job logs 'raysubmit_xxxxxxxxxxxxxxxx' --follow
+```
+
+#### Kubectl
+
+Observe the pods in your cluster with
+```shell
+# If you're on MacOS, first `brew install watch`.
+watch -n 1 kubectl get pod
+```
+
+#### Ray Dashboard
+
+View `localhost:8265` in your browser to access the Ray Dashboard.
+
+#### Ray Status
+
+Observe autoscaling status and Ray resource usage with
+```shell
+# Substitute the name of your Ray cluster's head pod.
+watch -n 1 kubectl exec -it raycluster-xgboost-benchmark-head-xxxxx -- ray status
+```
+
+:::{note}
+Under some circumstances and for certain cloud providers,
+the K8s API server may become briefly unavailable during Kuberentes
+cluster resizing events.
```

Comment:
Happened to me when running this workload on GKE and scaling from 1 to 10 nodes.
I think the explanation is that GKE decided to resize the K8s master.

Reply:
This temporary lack of availability was tolerated gracefully, which was nice.